docs/source/getting-started/app-architecture.rst

```diff
@@ -50,12 +57,16 @@ The last thing we'll point out about the DAML model for now is the operation to
   :end-before: -- ADDFRIEND_END
 
 DAML contracts are *immutable* (can not be changed in place), so the only way to "update" one is to archive it and create a new instance.
-That is what the ``AddFriend`` choice does: after checking some preconditions, it creates a new user contract with the new friend added to the list.
-The ``choice`` syntax automatically includes the archival of the current instance.
+That is what the ``AddFriend`` choice does: after checking some preconditions, it archives the current user contract and creates a new instance with the new friend added to the list.
+
+There is some boilerplate to set up the choice (full details in the :doc:`DAML reference </daml/reference/choices>`):
 
-.. TODO Update depending on consuming/nonconsuming choice.
+    - The choice is ``nonconsuming`` as we ``archive`` the contract explicitly.
```

Comment:
This sounds a bit confusing. Maybe something like “To make things explicit, we mark all choices in this guide as `nonconsuming` and call `archive` when we want to consume a contract”.